File: Problem 5/solution.js
// This Program finds the smallest positive number 
// that is evenly divisible by all of the numbers from 1 to 20

function gcd(a,b){
    if (b == 0){
        return a;
    }
    return gcd(b, a % b);
}

function lcm(a, b){
    return (a * b) / gcd(a, b);
}

var result = 1;

for (let i = 1; i <= 20; i++){
    result = lcm(result, i);  
} 

console.log(result);

File: Problem 5/solution.py
# This Program finds the smallest positive number 
# that is evenly divisible by all of the numbers from 1 to 20

def gcd(a,b):
    if b == 0:
        return a
    return gcd(b, a % b)

def lcm(a, b):
    return (a * b) // gcd(a, b)

result = 1

for i in range(2,21):
    result = lcm(result, i)    

print(result)